src/modules/commander/accelerometer_calibration.cpp

```diff
@@ -407,19 +405,22 @@ int do_accel_calibration(orb_advert_t *mavlink_log_pub)
 				accel_T_rotated.print();
 #endif // DEBUD_BUILD
 				calibrations[i].PrintStatus();
+				success = true;
 			}
 
 			// save all calibrations including empty slots
-			calibrations[i].ParametersSave();
+			if (!calibrations[i].ParametersSave()) {
+				calibration_log_critical(mavlink_log_pub, "calibration save failed");
+				success = false;
+				break;
+			}
 		}
 
-		param_notify_changes();
-
-		/* if there is a any preflight-check system response, let the barrage of messages through */
-		px4_usleep(200000);
-		calibration_log_info(mavlink_log_pub, CAL_QGC_DONE_MSG, sensor_name);
-		px4_usleep(600000); /* give this message enough time to propagate */
```

Comment:
From testing I don't see why this sleeping is still necessary. Preflight checks are now blocked during calibration.